dlib/test/dnn.cpp

```diff
@@ -2497,6 +2497,75 @@ namespace
 
 // ----------------------------------------------------------------------------------------
 
+    void test_loss_mean_squared_per_channel_and_pixel_simple()
+    {
+        print_spinner();
+
+        const int num_samples = 1000;
+        const long num_channels = 10;
+        const long dimension = 1000;
```

Comment:
This all looks great.  I suspect the reason it's still not converging super fast is just because these are large.  Try like 2 channels and dimensions of 3.